azurelinuxagent/common/protocol/wire.py

```diff
@@ -628,20 +623,22 @@ def fetch_manifest(self, version_uris):
             # in the lambda.
             host_func = lambda: self.fetch_manifest_through_host(version.uri)  # pylint: disable=W0640
 
-            try:
-                response = self.send_request_using_appropriate_channel(direct_func, host_func)
+            response = self.send_request_using_appropriate_channel(direct_func, host_func)
 
-                if response:
+            try:
+                if response is None:
+                    logger.warn("Failed to fetch manifest from {0}", version.uri)
+                else:
                     host = self.get_host_plugin()
                     host.update_manifest_uri(version.uri)
                     return response
-            except Exception as e:  # pylint: disable=C0103
-                logger.warn("Exception when fetching manifest. Error: {0}".format(ustr(e)))
+            except Exception as error:
+                logger.warn("Failed to fetch manifest from {0}. Error: {1}", version.uri, ustr(error))
 
         raise ExtensionDownloadError("Failed to fetch manifest from all sources")
 
     def stream(self, uri, destination, headers=None, use_proxy=None):
-        success = False
+        success = None
```

Comment:
what does None mean in this context? Is the value returned by this function tri-state? (None/False/True)

Reply:
also, previously stream was returning False, now it returns None.

Reply:
This change intentional, previously it was tri-state, now it's None/True.
Now all functions getting called in `send_request_using_appropriate_channel` return None on failure, not None or False.

Reply:
Discussed offline; reverted this change to keep interface consistent with current behaviour.